nntrainer/tensor/tensor.cpp

```diff
@@ -2956,33 +2945,39 @@ float Tensor::l2norm() const {
   NNTR_THROW_IF(!contiguous, std::invalid_argument)
     << getName() << " is not contiguous, cannot get l2norm.";
 
+  float ret = 0; // @todo fix ret to template
   unsigned int len = size();
   if (getDataType() == ml::train::TensorDim::DataType::FP32) {
     const float *data = getData<float>();
-    return snrm2(len, data, 1);
+    ret = snrm2(len, data, 1);
   } else if (getDataType() == ml::train::TensorDim::DataType::FP16) {
     const __fp16 *data = getData<__fp16>();
-    return snrm2(len, data, 1);
+    ret = snrm2(len, data, 1);
   }
+
+  return ret;
 }
 
 float Tensor::max_abs() const {
   NNTR_THROW_IF(!contiguous, std::invalid_argument)
     << getName() << " is not contiguous, cannot get max_abs.";
 
+  float ret = 0; // @todo fix ret to template
```

Comment:
```suggestion
  T ret = 0; 
```
ret should be template.